services/core-api/app/api/now_applications/resources/now_application_status_resource.py

```diff
@@ -0,0 +1,119 @@
+            raise NotFound('No permit amendment found for this application.')
+
+        if now_application_status_code is not None and now_application_identity.now_application.now_application_status_code != now_application_status_code:
+            now_application_identity.now_application.status_updated_date = datetime.today()
+            now_application_identity.now_application.now_application_status_code = now_application_status_code
+
+            # Approved
+            if now_application_status_code == 'AIA':
+                #move out of draft
+                if permit.permit_status_code == 'D':
+                    permit.permit_status_code = 'O'
+
+                #assign permit_no
+                permit.assign_permit_no(
+                    now_application_identity.now_application.notice_of_work_type_code[0])
+
+                permit.save()
+
+                permit_amendment.permit_amendment_status_code = 'ACT'
+                permit_amendment.issue_date = issue_date
+                permit_amendment.auth_end_date = auth_end_date
+                permit_amendment.save()
+
+                #create contacts
+                for contact in now_application_identity.now_application.contacts:
+
+                    new_permittee = False
+
+                    if contact.mine_party_appt_type_code == "PMT":
+                        current_mpa = MinePartyAppointment.find_current_appointments(
+                            mine_party_appt_type_code=contact.mine_party_appt_type_code,
+                            permit_id=permit.permit_id)
+                        if len(current_mpa) > 1:
+                            raise BadRequest(
+                                'This permit has more than one active permittee. Please resolve this and try again.'
+                            )
+                        if len(current_mpa) == 1:
+                            if current_mpa[0].party_guid != contact.party_guid:
+                                current_mpa[0].end_date = start_date - timedelta(days=1)
+                                current_mpa[0].save()
+                                new_permittee = True
+
+                    if contact.mine_party_appt_type_code != "PMT" or new_permittee:
+                        mine_party_appointment = MinePartyAppointment.create(
+                            mine=now_application_identity.mine
+                            if contact.mine_party_appt_type_code != 'PMT' else None,
+                            permit=permit if contact.mine_party_appt_type_code == 'PMT' else None,
+                            party_guid=contact.party_guid,
+                            mine_party_appt_type_code=contact.mine_party_appt_type_code,
+                            start_date=datetime.utcnow(),
+                            end_date=None,
+                            processed_by=self.get_user_info())
+                        mine_party_appointment.save()
+
+                #TODO: Documents / CRR
```

Comment:
Is this TODO still relevant?

Reply:
Yes, it's probably a followup ticket. I could make some assumptions but there was ambiguity over what should happen with documents, and we don't have spots for all of them. "Permit required reports" versus "Code required reports"